File: message-board-api/app/models/message.rb
class Message < ApplicationRecord
    belongs_to :sender, :class_name => 'User', :foreign_key => 'sender_id', required: false
    belongs_to :recipient, :class_name => 'User', :foreign_key => 'recipient_id', required: false

    validates :content, presence: true

    accepts_nested_attributes_for :sender, :recipient, reject_if: proc { |attributes| attributes['name'].blank? }
end

File: client/src/components/Message.js
import React from 'react';

const Message = ({ message }) => {
  const sender = message.sender;
  const recipient = message.recipient;

  return (
    <div className="message-card">
        { (sender || recipient) ?
            <div className="message-card-header">
              {sender ? <span className="sender">From: {message.sender.name}</span> : null }

              {recipient ? <span className="recipient">To: {message.recipient.name}</span> : null }
            </div>
            : null
        }

        <div className="message-card-main">
          {message.content}
        </div>
    </div>
  );
};

export default Message;